### DISCUSSION TITLE
Load app in IRB (like `rails console`)

### DISCUSSION THREAD

**Discussion by davekaro** [jeremyevans/roda#263] [General]
I've been playing with Roda and curious how folks enter a quick "console session" like `rails console` provides. I've been moving `require` statements around between `config.ru` and `app.rb` and testing different strategies with Zeitwerk and rack-unreloader.

I thought easiest would be to make `app.rb` load everything, then I added an `.irbrc` file with `require_relative "app"` and it worked. However, now that I have more requires in `app.rb`, I can't follow the advice at the bottom of https://github.com/jeremyevans/rack-unreloader#dependency-handling- to load dependencies in `config.ru` so they aren't reloaded every time `app.rb` changes.

Curious how others handle this. Thanks!

**Comment by jeremyevans** [comment]
It seem like you could have your `.irbrc` file load the dependencies that `config.ru` loads before loading `app.rb`.  Is that possible?

I haven't tried to do this in my own apps.  Generally for an irb shell, I just load the models (`irb -r ./models`), and I don't have the irb shell do reloading.

**Reply by davekaro** [reply]
Ah yes that makes sense. I was stuck thinking I wanted everything loaded with one require. 

You have a good point about just loading the models since most IRB sessions are interacting with those. Hard to get my head out of the sand after blindly following rails conventions for years ;)

**Comment by adam12** [comment]
I'm not sure if this would still work, but at one point in time [racksh](https://rubygems.org/gems/racksh) was fairly popular. I looked over the source and cant' see any reason why it wouldn't still work today.

**Comment by davekaro** [comment]
racksh looks neat. I also found the `Rakefile` from the roda-sequel-stack includes a nice task `rake dev_irb` - see https://github.com/jeremyevans/roda-sequel-stack/blob/f3d555f978c4bc56e38ddfea643faabaeccea2d7/Rakefile#L69
